ocean_lib/ocean/ocean_assets.py

```diff
@@ -322,48 +375,88 @@ def create(
             logger.error(msg)
             raise ValueError(msg)
 
-        # Process the DDO
-        asset_dict = asset.as_dictionary()
-        ddo_string = json.dumps(asset_dict, separators=(",", ":"))
-        ddo_bytes = ddo_string.encode("utf-8")
-        ddo_hash = create_checksum(ddo_string)
+        document, flags, ddo_hash = self.process_ddo(
+            asset, provider_uri, encrypt_flag, compress_flag
+        )
 
-        # Plain asset
-        if not encrypt_flag and not compress_flag:
-            flags = bytes([0])
-            document = ddo_bytes
+        _ = erc721_token.set_metadata(
```

Comment:
if you don't need the assignment, then don't assign it. We use the _, something assignment when we have a tuple return type, to get "half" of the return, but in this case you don't need to assign at all, if the only return value is unused